modules/virtual_network_gateway/main.tf

```diff
@@ -122,18 +125,18 @@ data "azurerm_public_ip" "exists" {
 resource "azurerm_local_network_gateway" "this" {
   for_each = var.local_network_gateways
 
-  name                = "${var.name_prefix}lgw-vgw${var.name_suffix}-${each.value.name}"
+  name                = each.value.local_ng_name
   resource_group_name = var.resource_group_name
   location            = var.location
   gateway_address     = each.value.gateway_address
-  address_space       = try(each.value.address_space, [])
+  address_space       = each.value.address_space
 
   dynamic "bgp_settings" {
     for_each = each.value.remote_bgp_settings
```

Comment:
this is defined as an optional list, w/o any default value, hence null - this will fail -> add a default value of [] to type

Reply:
added